templates/crate/source.html

```diff
@@ -127,8 +127,15 @@
                 {% else %}
                     {% set file_name = "" %}
                 {% endif %}
-                <div id="source-code" class="pure-u-1 pure-u-sm-17-24 pure-u-md-19-24">
-                    {{- file_content|highlight(file_name)|safe -}}
+                <div id="source-code-container" class="pure-u-1 pure-u-sm-17-24 pure-u-md-19-24">
+                    <div data-nosnippet class="source-code"><pre id="line-numbers"><code>
+                        {%- for line in 1..=file_content.lines().count() -%}
+                            <a href="#{{line|safe}}" id="{{line|safe}}">{{line|safe}}</a>
+                        {%~ endfor -%}
```

Comment:
What is the `~` doing?

Reply:
The `~` keeps one white space character. Otherwise, it seems to remove the line breaks as well and all line numbers are placed on the same line.

Reply:
Oh I see. Then the `<code>` should be removed but the `~` should be kept.

Reply:
The `<code>` tag was intentional, as some css rules target `pre > code`. I can remove the `<code>` tag, but that would mean that some rules need to be duplicated. I tried to change the rules to simply target `pre` but this doesn't work directly and needs more adaptation.

So, shall I remove the tag or do we want to keep it?

Reply:
Hum... Let's go to the simplest then and keep the `<code>` then.